timemachine/cpp/src/tibd_exchange_move.cu

```diff
@@ -171,10 +169,42 @@ void TIBDExchangeMove<RealType>::move(
     curandErrchk(
         templateCurandUniform(this->cr_rng_samples_, this->d_sample_noise_.data, this->d_sample_noise_.length));
     k_generate_translations_inside_and_outside_sphere<<<1, d_rand_states_.length, 0, stream>>>(
-        this->steps_per_move_, d_box, d_center_.data, radius_, d_rand_states_.data, this->d_translations_.data);
+        this->num_proposals_per_move_, d_box, d_center_.data, radius_, d_rand_states_.data, this->d_translations_.data);
     gpuErrchk(cudaPeekAtLastError());
 
-    for (int step = 0; step < this->steps_per_move_; step++) {
+    /* --Algorithm Description--
+    * Targeted Insertion Biased Deletion algorithm is as follows
+    *
+    * 1. Generate all random noise upfront to ensure bitwise identical results regardless of batch size
+    * 2. Compute the initial weights of each of the molecules (no batching)
+    * 3. Copy the initial weights (d_log_weights_before_) to the proposal weight buffers (d_log_weights_after_),
+    *    duplicating the values for each proposal in the batch
+    * 4. Flag the target molecules that are inside and outside the target region
+    * 5. Set up the proposals determining if they should be targeting the inner region or the outer region
+    * 6. Construct the offsets for the initial weights (d_src_log_weights_) and after (d_log_weights_after_) log weights
+    * 7. Separate out the weights (targeting inner region only looks at outer mols) associated with each proposal
+    * 8. For each proposal in the batch sample a weight index from the initial weights, aiming to select indexes with high energies
+    * 9. Compute the logexpsum (using SegmentedSumExp and compute_logsumexp_final) of each proposal's initial weights.
+    * 10. Remap the indexes selected in 8. back to molecule indexes to be able to determine which coordinates to modify in the next
+    *     step.
+    * 11. Generate the proposals for all of the sampled molecules in the batch, rotating and translating the mols to the new
+    *    positions.
+    * 12. For each proposal in the batch separate out the proposals weights into the after log weights.
+    * 13. Compute the logexpsum (using SegmentedSumExp and compute_logsumexp_final) of each proposal's after log weights.
+    * 14. Find the first proposal in the batch that was accepted with the metropolis hasting check
```

Comment:
Metropolis-Hastings

Reply:
Fixed up all instances of this style of Metropolis-Hastings across our C++ code: https://github.com/proteneer/timemachine/pull/1283/commits/05248f9ed66b4b924d9c339c8ff24d30d9d55931